server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -257,6 +259,34 @@ public static List<String> getWordList(
         }
     }
 
+    public static List<String> getWordList(
+        Environment env,
+        Settings settings,
+        String settingPath,
+        String settingList,
+        boolean removeComments,
+        boolean checkDuplicate
+    ) {
```

Comment:
This new method might be misleading since we check duplicates only on the first split (before the first space in the rule). I'd prefer a separate method something like `removeDuplicateRules` with a clear comment explaining how we check for duplicates.